src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1114,28 +1115,21 @@
   "Prep `event` for comparison/computation of a hash"
   [event]
   (-> event
-      (utils/update-when [:timestamp] to-timestamp)
-      (utils/update-when [:old_value] sutils/db-serialize)
-      (utils/update-when [:new_value] sutils/db-serialize)
+      (update :timestamp to-timestamp)
+      (update :old_value sutils/db-serialize)
+      (update :new_value sutils/db-serialize)
       (assoc :containing_class (find-containing-class (:containment_path event)))))
 
 (defn normalize-report
   "Prep the report for comparison/computation of a hash"
-  [report]
+  [{:keys [resources] :as report}]
   (-> report
-      (update-in [:start_time] to-timestamp)
-      (update-in [:end_time] to-timestamp)
-      (update-in [:producer_timestamp] to-timestamp)
-      (update-in [:resource_events] #(map normalize-resource-event %))))
-
-(defn convert-containment-path
-  "Convert the contain path from a collection to the jdbc array type"
-  [event]
-  (utils/update-when event
-                     [:containment_path]
-                     (fn [cp]
-                       (when cp
-                         (sutils/to-jdbc-varchar-array cp)))))
+      (update :start_time to-timestamp)
+      (update :end_time to-timestamp)
+      (update :producer_timestamp to-timestamp)
+      (assoc :resource_events (->> resources
```

Comment:
I think we should definitely not do that, but for now this seems ok because moving to using the resources field would require significant test/testutils/examples changes, even more than are already here.